qcodes/tests/dataset/helper_functions.py

```diff
@@ -73,14 +81,18 @@ def verify_data_dict_for_single_param(datadict: Dict[str, np.ndarray],
         assert_array_equal(mydata, value)
 
 
-def verify_dataframe_for_single_param(dataframe: pandas.DataFrame,
-                                      names: Sequence[str],
-                                      shapes: Sequence[Tuple[int, ...]],
-                                      values):
+def verify_dataframe_for_single_param(
+    dataframe: pd.DataFrame,
+    names: Sequence[str],
+    shapes: Sequence[tuple[int, ...]],
+    values,
+):
+    import pandas as pd
```

Comment:
```suggestion
    import pandas as pd  # pylint: disable=import-outside-toplevel
```
(and similar at other locations)